### PR TITLE
Issue #1161 metaswap regrid model

### LINKED ISSUE #1161: add regrid_like method to metaswap model
-should regrid most of its packages, like MeteoMapping, Infiltration , ...

-should create  new instances of  other packages, like IDFMappig, CouplerMapping

-should accept regridder settings to specify how to regrid the arrays in the packages.

### PR DESCRIPTION
**Pull request by luitjansl**
Fixes #1161 

# Description
Implemented a regrid_like method for metaswap models

# Checklist

- [x] Links to correct issue
- [ ] Update changelog, if changes affect users
- [x] PR title starts with ``Issue #nr``, e.g. ``Issue #737``
- [x] Unit tests were added
- [ ] **If feature added**: Added/extended example

### CODE REVIEW

File: imod/msw/model.py [hunk 1]
```diff
@@ -242,3 +245,34 @@ def write(self, directory: Union[str, Path]):
...
+        regrid_context: Optional[RegridderWeightsCache] = None,
+        regridder_types: Optional[dict[str, Tuple[RegridderType, str]]] = None,
+    ) -> "MetaSwapModel":
+        unsat_database = self.simulation_settings["unsa_svat_path"]
+        regridded_model = MetaSwapModel(unsat_database)
+
+        target_grid = mf6_regridded_dis["idomain"]
+
+        mod2svat = self.pop("mod2svat", None)
+        grid = self.pop("grid", None)
```
Comment (JoerivanEngelen) [inline imod/msw/model.py:262]: These keys are not hardcoded, but decided by the user. I think you have to find packages based on their type and pop them I fear...

File: imod/msw/sprinkling.py [hunk 2]
```diff
@@ -94,3 +94,6 @@ def _render(self, file, index, svat):
         self._check_range(dataframe)
 
         return self.write_dataframe_fixed_width(file, dataframe)
+
+    def is_regridding_supported(self) -> bool:
+        return False
```
Comment (JoerivanEngelen) [inline imod/msw/sprinkling.py:99]: Please add a comment that this can be changed when this is resolved https://github.com/Deltares/imod-python/issues/728

**Review by JoerivanEngelen — APPROVED** [review]
2 comments!

File: imod/msw/sprinkling.py (hunk 2 above)
Comment (luitjansl) [reply imod/msw/sprinkling.py:99]: done

File: imod/msw/model.py (hunk 1 above)
Comment (luitjansl) [reply imod/msw/model.py]: done

**Pull request merged**